### PR TITLE
[202305] Optimize syslog rate limit feature for fast and warm boot

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
Backport PR https://github.com/sonic-net/sonic-buildimage/pull/17458 due to conflict.

#### Why I did it

Optimize syslog rate limit feature for fast and warm boot

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it

1. Optimize redis start time
2. Don't render rsyslog.conf in container startup script
3. Disable containercfgd by default. There is a new CLI to enable it (in another PR)

#### How to verify it

Manual test
Regression test

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by keboliu — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@saiarcot895 would you please help to review as well?

**Comment by liat-grozovik** [comment]
@saiarcot895 @StormLiangMS can you please help to merge?

File: files/build_templates/docker_image_ctl.j2 [hunk 1]
```diff
@@ -326,6 +315,12 @@ start() {
     # Obtain our platform as we will mount directories with these names in each docker
     PLATFORM=${PLATFORM:-`$SONIC_CFGGEN -H -v DEVICE_METADATA.localhost.platform`}
 
+    # Default rsyslog target IP for single ASIC platform
+    SYSLOG_TARGET_IP=127.0.0.1
+    if [[ ($NUM_ASIC -gt 1) ]]; then
+        SYSLOG_TARGET_IP=$(docker network inspect bridge --format={{ "'{{(index .IPAM.Config 0).Gateway}}'" }})
+    fi
```
Comment (saiarcot895) [inline files/build_templates/docker_image_ctl.j2:322]: This needs the multi-ASIC fix here.

Comment (Junchao-Mellanox) [reply files/build_templates/docker_image_ctl.j2]: Fixed

**Comment by Junchao-Mellanox** [comment]
The checker failure is pending PR https://github.com/sonic-net/sonic-buildimage/pull/17486 to be cherry-picked to 202305.

**Comment by StormLiangMS** [comment]
hi @Junchao-Mellanox this PR change a lot files, could you have a full qual with 202305 image?

**Comment by Junchao-Mellanox** [comment]
Hi, @StormLiangMS , what kind of qual do you need? We have daily regression on top of this change with all topos we support.

**Comment by Junchao-Mellanox** [comment]
Tested on sonic-buildimage 202305 commit hash 5b1d18898

**Comment by saiarcot895** [comment]
@Junchao-Mellanox Do you have numbers on how much time this PR saves?

**Comment by Junchao-Mellanox** [comment]
Hi @saiarcot895 , I think it is platform specific. We tested it on mellanox 2700 platform, it saves ~4 seconds.

**Comment by StormLiangMS** [comment]
Hi @qiluo-msft could you help to take a look?

**Pull request merged**